crates/sui-core/src/mock_consensus.rs

```diff
@@ -96,18 +96,27 @@ impl MockConsensusClient {
             }
         }
     }
+
+    fn submit_impl(&self, transactions: &[ConsensusTransaction]) -> SuiResult<BlockStatusReceiver> {
+        // TODO: maybe support multi-transactions and remove this check
+        assert!(transactions.len() == 1);
+        let transaction = &transactions[0];
+        self.tx_sender
+            .try_send(transaction.clone())
+            .expect("MockConsensusClient channel should not overflow");
```

Comment:
why is this safe? couldn't we fill up the channel if we submitted too much too quickly?

Reply:
Yes, but it's a mock only used in tests that don't do that?